src/ffi/sync/builder.rs

```diff
@@ -41,18 +43,24 @@ impl Builder {
         result!(addr, Builder { addr, device })
     }
 
-    pub fn add_optimization_profile(&mut self) -> Result<()> {
+    pub fn create_optimization_profile(&mut self) -> Result<OptimizationProfile> {
         let internal = self.as_mut_ptr();
         let optimization_profile_internal = cpp!(unsafe [
             internal as "void*"
         ] -> *mut std::ffi::c_void as "void*" {
             return ((IBuilder*) internal)->createOptimizationProfile();
         });
-        result!(optimization_profile_internal)
+        let profile = result!(optimization_profile_internal, optimization_profile_internal)?;
+        Ok(OptimizationProfile::wrap(profile))
```

Comment:
Since the TensorRT docs state that the lifetime of the optimisation profile is bound to the builder object. Wrapping it here disjoins the lifetimes which is unsafe unfortunately. To better model the TensorRT API, we should add the optimisation profile to an internal Vec or model it transiently the same way we're doing it with `Tensor` in network (with accessor functions).